common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -92,15 +101,32 @@
                 AuthorizationResult> {
 
     private static final String TAG = MicrosoftStsOAuth2Strategy.class.getSimpleName();
+    private IDevicePopManager mDevicePopManager;
 
     /**
      * Constructor of MicrosoftStsOAuth2Strategy.
      *
-     * @param config MicrosoftStsOAuth2Configuration
+     * @param config  MicrosoftStsOAuth2Configuration
+     * @param options OAuth2StrategyOptions
      */
-    public MicrosoftStsOAuth2Strategy(@NonNull final MicrosoftStsOAuth2Configuration config) {
-        super(config);
-        setTokenEndpoint(config.getTokenEndpoint().toString());
+    public MicrosoftStsOAuth2Strategy(@NonNull final MicrosoftStsOAuth2Configuration config,
+                                      @NonNull final OAuth2StrategyOptions options) {
+        super(config, options);
+        // TODO figure this out!
```

Comment:
TODO

Where can this move / how should this be structured to allow these exceptions to be thown/caught in a checked manner?

Reply:
This has been [somewhat] improved upon in 679aab6eb3368ed87e403b5acbc3ec0fdf4857ef

Now, the strategy's constructor will throw an exception up to the authority which constructs it. The authority will, in turn, throw the exception up to the calling controller who will shuttle the error back to the error callback